src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs

```diff
@@ -292,6 +301,26 @@ internal static SecurityKey GetSecurityKey(EncryptingCredentials encryptingCrede
 
                 securityKey = encryptingCredentials.Key;
             }
+#if NET472
+            else if (SupportedAlgorithms.EcdsaWrapAlgorithms.Contains(encryptingCredentials.Alg))
+            {
+                // on decryption we get the public key from the EPK value see: https://datatracker.ietf.org/doc/html/rfc7518#appendix-C
+                string apu = null, apv = null;
+                if (additionalHeaderClaims != null && additionalHeaderClaims.Count > 0)
+                {
+                    if (additionalHeaderClaims.TryGetValue(JwtHeaderParameterNames.Apu, out object objApu))
+                        apu = objApu.ToString();
+                    if (additionalHeaderClaims.TryGetValue(JwtHeaderParameterNames.Apv, out object objApv))
+                        apv = objApv.ToString();
+                }
+                EcdhKeyExchangeProvider ecdhKeyExchangeProvider = new EcdhKeyExchangeProvider(encryptingCredentials.Key as ECDsaSecurityKey, encryptingCredentials.JsonWebKey, encryptingCredentials.Alg, encryptingCredentials.Enc);
+                SecurityKey kdf = ecdhKeyExchangeProvider.GenerateKdf(apu, apv);
+                kwProvider = cryptoProviderFactory.CreateKeyWrapProvider(kdf, ecdhKeyExchangeProvider.GetEncryptionAlgorithm());
+                SecurityKey cek = new SymmetricSecurityKey(JwtTokenUtilities.GenerateKeyBytes(ecdhKeyExchangeProvider.KeyDataLen * 2));
```

Comment:
>ecdhKeyExchangeProvider.KeyDataLen * 2 [](http://example.com/codeflow?start=94&length=38)

If ecdhKeyExchangeProvider.KeyDataLen is size in bits, what is the logic of 2x? This seems like a magic number and is not clear.